test/index.test.js

```diff
@@ -937,6 +939,23 @@ describe('test fetch retry on http errors (throw exceptions)', () => {
         }
         assert(nock.isDone());
     });
+    it('test network timeout before while loop (in isResponseTimedOut)', async () => {
+        const rewiredFetchRetry = rewire('../index');
+        const getTimeRemainingObject = {getTimeRemaining: rewiredFetchRetry.__get__('getTimeRemaining')};
+
+        const getTimeRemainingStub = sinon.stub(getTimeRemainingObject, 'getTimeRemaining').callsFake(() => {
+            console.log("Mocked isResponseTimedOut");
+            return -1;
+        });
+        rewiredFetchRetry.__set__('getTimeRemaining', getTimeRemainingStub);
+        try {
+            await rewiredFetchRetry(`${FAKE_BASE_URL}${FAKE_PATH}`, { method: 'GET' });
+            assert.fail("Should have thrown an error!");
+        } catch (e) {
+            assert(e.message.includes("network timeout"));
```

Comment:
Ack - I know this is the style we did those tests. But a good change would also to check the expected exception using something like `assert.throws(...)`.